IdentityCore/tests/automation/ui_tests_lib/lab_api/MSIDAutomationOperationAPIRequestHandler.m

```diff
@@ -81,25 +87,33 @@ - (void)getAccessTokenAndCallLabAPI:(MSIDAutomationBaseApiRequest *)request
                     responseHandler:(id<MSIDAutomationOperationAPIResponseHandler>)responseHandler
                   completionHandler:(void (^)(id result, NSError *error))completionHandler
 {
+    NSData *base64EncodedCert = [[NSData alloc] initWithBase64EncodedString:self.encodedCertificate options:0];
+    if (!base64EncodedCert || base64EncodedCert.length == 0) {
+        NSLog(@"Couldn't fetch certificate data, make sure certificate path is correct");
+        return;
+    }
+    
     [MSIDClientCredentialHelper getAccessTokenForAuthority:self.configurationParams[@"operation_api_authority"]
                                                   resource:self.configurationParams[@"operation_api_resource"]
                                                   clientId:self.configurationParams[@"operation_api_client_id"]
-                                          clientCredential:self.configurationParams[@"operation_api_client_secret"]
-                                         completionHandler:^(NSString *accessToken, NSError *error) {
-                                             
-                                             if (!accessToken)
-                                             {
-                                                 dispatch_async(dispatch_get_main_queue(), ^{
-                                                     completionHandler(nil, error);
-                                                 });
-                                                 return;
-                                             }
-                                             
-                                             [self executeAPIRequestImpl:request
-                                                         responseHandler:responseHandler
-                                                             accessToken:accessToken
-                                                       completionHandler:completionHandler];
-                                         }];
+                                               certificate:base64EncodedCert 
+                                       certificatePassword:self.certificatePassword
```

Comment:
where do we read the certificate password from ?

Reply:
It will be read from the conf.json file, which we stored in the key_vault. The json file is not checked in here.